app/src/src_nonfreenet/org/breezyweather/sources/smg/SmgService.kt

```diff
@@ -0,0 +1,422 @@
+
+    // Even in its English bulletins, SMG refers to itself as "SMG" rather than
+    // "Macao Meteorological and Geophysical Bureau". Keep the Portuguese name
+    // for use in the source list. The English name can be used in attributions.
+    override val name by lazy {
+        if (context.currentLocale.code.startsWith("zh")) {
+            "地球物理氣象局"
+        } else {
+            "Serviços Meteorológicos e Geofísicos (SMG)"
+        }
+    }
+    override val privacyPolicyUrl by lazy {
+        with(context.currentLocale.code) {
+            when {
+                startsWith("zh") -> "https://www.smg.gov.mo/zh/subpage/21/page/21"
+                startsWith("pt") -> "https://www.smg.gov.mo/pt/subpage/21/page/21"
+                else -> "https://www.smg.gov.mo/en/subpage/21/page/21"
+            }
+        }
+    }
+
+    override val color = Color.rgb(1, 173, 159)
+    override val weatherAttribution by lazy {
+        with(context.currentLocale.code) {
+            when {
+                startsWith("zh") -> "地球物理氣象局"
+                startsWith("pt") -> "Direcção dos Serviços Meteorológicos e Geofísicos"
+                else -> "Macao Meteorological and Geophysical Bureau"
+            }
+        }
+    }
+
+    private val mApi by lazy {
+        client
+            .baseUrl(SMG_BASE_URL)
+            .build()
+            .create(SmgApi::class.java)
+    }
+
+    private val mCmsApi by lazy {
+        client
+            .baseUrl(SMG_CMS_URL)
+            .build()
+            .create(SmgCmsApi::class.java)
+    }
+
+    override val supportedFeaturesInMain = listOf(
+        SecondaryWeatherSourceFeature.FEATURE_CURRENT,
+        SecondaryWeatherSourceFeature.FEATURE_ALERT,
+        SecondaryWeatherSourceFeature.FEATURE_AIR_QUALITY,
+        SecondaryWeatherSourceFeature.FEATURE_NORMALS
+    )
+
+    override fun isFeatureSupportedInMainForLocation(
+        location: Location,
+        feature: SecondaryWeatherSourceFeature?,
+    ): Boolean {
+        return location.countryCode.equals("MO", ignoreCase = true)
+    }
+
+    override fun requestWeather(
+        context: Context,
+        location: Location,
+        ignoreFeatures: List<SecondaryWeatherSourceFeature>,
+    ): Observable<WeatherWrapper> {
+        val lang = when (context.currentLocale.code.substring(0, 2)) {
+            "zh" -> "c"
+            "pt" -> "p"
+            else -> "e"
+        }
+
+        val daily = mApi.getDaily()
+        val hourly = mApi.getHourly()
+
+        // ASTRO
+        val formatter = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
+        formatter.timeZone = TimeZone.getTimeZone("Asia/Macau")
+        val now = Calendar.getInstance(TimeZone.getTimeZone("Asia/Macau"))
+        var body: String
+        val astroList = MutableList<Observable<SmgAstroResult>>(8) {
+            body = """
+                {"date":"${formatter.format(now.time)}"}
+            """.trimIndent()
+            now.add(Calendar.DATE, 1)
```

Comment:
Overkill. MET Norway has a similar issue (one call = one day), but I only call it once (today). Other days are a minor feature in a sub activity of the app. For that case, we can let the app calculate it by itself.

Reply:
Kept just one.